### ISSUE TITLE
Question on Electricity Prices

### ISSUE THREAD

**Issue by thompsoa** [JGCRI/gcam-core#354]
Hi, I'm trying to understand the impact of different scenarios on revenues and costs in the utilities sector, and I'm not quite sure how to think about sequestration. In some 1.5C scenarios, or even 2C scenarios, there can a lot of carbon capture in the model, leading to very negative net emissions for electricity. Naively, multiplying net emissions by carbon price would imply a large subsidy to the sector, is this actually what is happening in the model? 

Specifically, how are final electricity prices set, and how do carbon prices impact on them? Is there a fixed passthrough of costs? Do negative emissions generate a subsidy to offset the costs of installing CCS? and is this also passed through as lower prices? Is there a document that lays out the exact equations without trying to work through the code? 

Thanks!

**Comment by pkyle** [comment]
It's a pretty simple representation, and I don't know that it's written anywhere specifically, but the cost of any technology is printed to the output database and there's a query ("elec gen costs by tech"). The total technology cost can be computed off-line as the sum of:
* non-energy cost (which itself is the sum of levelized capital cost, fixed O&M, and variable non-fuel O&M)
* energy costs (sum of each input-output coefficient times the respective input price)
* emissions penalties (co2 price times the co2 intensity of the technology)
The non-energy costs are read to the model as differentiated capital, fixed O&M, and variable non-fuel O&M costs. However they are levelized by the model already, if you print out the query "costs by tech and input". The CO2 price is 1990$/tC, so that needs to be converted both to a common dollar year, and perhaps to CO2, though note that fuel carbon intensities are indicated as kgC/GJ (Ccoef.xml). So, emissions penalties are estimated as CO2 prices times the sum of each input's carbon intensity times its carbon intensity, making deductions for CO2 stored as appropriate. BECCS sequestration isn't credited to the power sector, but rather to the biomass production sector ("regional biomass") which converts atmospheric CO2 into an energy commodity, which can have a negative price. The BECCS technologies in the power sector are represented as having positive CO2 emissions (from the carbon in the fuels that isn't captured, about 5% or 10% I think). But the costs could hypothetically be negative if the biomass prices are negative enough.
I'm not sure if technologies with net negative costs are allowed in the power sector right now, and I don't have any recent model runs on hand with high enough carbon prices to trigger net-negative costs, but it sounds like you might be able to query that in your database and see.

**Comment by raditya101010** [comment]
Hi,

I wonder, has the "elec gen costs by tech" of each technology (the electricity one) already included the non-energy and energy costs?

Another question related to this thread, I am still not clear on how the electricity price is calculated in GCAM. I have been looking it up in the documentation, but couldn't find one. Could you explain that please?

Thanks.

**Comment by pkyle** [comment]
Yes the query "elec gen costs by tech" includes the net total of all cost components, positive and negative, for any generation technology. Computing that in a spreadsheet for any specific technology can be somewhat challenging, but it can be done, just by calculating all the cost components that I think I identified above, performing necessary unit conversions, adding them up. From that point, the remainder of the cost calculations are quite straightforward. The generation costs by subsector are equal to the weighted average of the technologies within each subsector, and similarly, the price of the "electricity" commodity is equal to the weighted average cost of all technologies (or equivalently subsectors). Note that hydroelectricity is modeled as a "fixed output" technology, and as such is not included in any of these price calculations.
The next stage in the electricity flow is deducting "own use", or electricity used on site, upstream of the transmission and distribution network. This is usually just a few percent, but it does have the effect of increasing the ultimate price of electricity paid by end-users.
From here, `electricity_net_ownuse` goes to the transmission and distribution network, where there are three sectors for the different end use classes (`bld`, `ind`, `trn`). All have the same distribution loss coefficient, but the cost adders are differentiated, allowing the industrial sector to see lower prices of purchased electricity then are seen in the buildings and transportation sectors. As with own use, distribution losses similarly contribute to hire end user prices.
The one final adjustment is in the building sector, where rooftop PV competes with grid electricity; the price of the final commodity `elect_td_bld` is the weighted average of self generated PV and purchased grid electricity.
